metadata-ingestion/src/datahub/ingestion/source/unity/proxy_profiling.py

```diff
@@ -0,0 +1,192 @@
+    ColumnProfile,
+    TableProfile,
+    TableReference,
+)
+from datahub.ingestion.source.unity.report import UnityCatalogReport
+from datahub.utilities.lossy_collections import LossyList
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+# TODO: Move to separate proxy/ directory with rest of proxy code
+class UnityCatalogProxyProfilingMixin:
+    _workspace_client: WorkspaceClient
+    _unity_catalog_api: UnityCatalogApi
+    report: UnityCatalogReport
+    warehouse_id: str
+
+    def check_profiling_connectivity(self):
+        self._workspace_client.warehouses.get(self.warehouse_id)
+        return True
+
+    def start_warehouse(self) -> Optional[Wait[GetWarehouseResponse]]:
+        """Starts a databricks SQL warehouse.
+
+        Returns:
+            - A Wait object that can be used to wait for warehouse start completion.
+            - None if the warehouse does not exist.
+        """
+        try:
+            return self._workspace_client.warehouses.start(self.warehouse_id)
+        except DatabricksError as e:
+            logger.warning(f"Unable to start warehouse -- are you sure it exists? {e}")
+            return None
+
+    def get_table_stats(
+        self, ref: TableReference, max_wait_secs: int, call_analyze: bool
+    ) -> Optional[TableProfile]:
+        """Returns profiling information for a table.
+
+        Performs three steps:
+        1. Call ANALYZE TABLE to compute statistics for all columns
+        2. Poll for ANALYZE completion with exponential backoff, with `max_wait_secs` timeout
+        3. Get the ANALYZE result via the properties field in the tables API.
+            This is supposed to be returned by a DESCRIBE TABLE EXTENDED command, but I don't see it.
+
+        Raises:
+            DatabricksError: If any of the above steps fail
+        """
+
+        # Currently uses databricks sdk, which is synchronous
+        # If we need to improve performance, we can manually make requests via aiohttp
+        try:
+            if call_analyze:
+                response = self._analyze_table(ref)
+                success = self._check_analyze_table_statement_status(
+                    response, max_wait_secs
+                )
+                if not success:
+                    self.report.profile_table_timeouts.append(str(ref))
+                    return None
+            return self._get_table_profile(ref)
+        except DatabricksError as e:
+            self.report.profile_table_errors.setdefault(str(e), LossyList()).append(
+                str(ref)
+            )
+            logger.warning(
+                f"Failure during profiling {ref}: ({e.error_code}) {e}", exc_info=True
+            )
+            return None
+
+    def _analyze_table(self, ref: TableReference) -> ExecuteStatementResponse:
+        response = self._workspace_client.statement_execution.execute_statement(
+            statement=f"ANALYZE TABLE {ref.schema}.{ref.table} COMPUTE STATISTICS FOR ALL COLUMNS",
```

Comment:
This is the column level stats. Do we want to support table-level stats only as well? 
I think later it would make sense as it is a way less expensive operation.

Reply:
I'll do this as a followup.